corehq/motech/repeaters/dbaccessors.py

```diff
@@ -74,34 +77,41 @@ def get_overdue_repeat_record_count(overdue_threshold=datetime.timedelta(minutes
     return results['value'] if results else 0
 
 
-def _get_startkey_endkey_all_records(domain, repeater_id=None, state=None):
+def _get_startkey_endkey_all_records(domain, repeater_id=None, state=None, datespan=None):
     kwargs = {}
 
-    if repeater_id and not state:
-        kwargs['endkey'] = [domain, repeater_id]
-        kwargs['startkey'] = [domain, repeater_id, {}]
-    elif repeater_id and state:
+    if datespan:
+        # reversed because descending:True
+        enddatekey = json_format_datetime(datespan.startdate) if datespan.startdate else ""
+        startdatekey = json_format_datetime(datespan.enddate) if datespan.enddate else ""
+        kwargs['endkey'] = [domain, repeater_id, state, enddatekey]
+        kwargs['startkey'] = [domain, repeater_id, state, startdatekey]
+    elif state:
         kwargs['endkey'] = [domain, repeater_id, state]
-        kwargs['startkey'] = [domain, repeater_id, state, {}]
-    elif not repeater_id and state:
-        kwargs['endkey'] = [domain, None, state]
-        kwargs['startkey'] = [domain, None, state, {}]
-    elif not repeater_id and not state:
-        kwargs['endkey'] = [domain, None]
-        kwargs['startkey'] = [domain, None, {}]
+        kwargs['startkey'] = [domain, repeater_id, state, None]
```

Comment:
why is this `None`? I think it should still be `{}`

Reply:
Sure, I change it back. But I would like to understand why `{}` and not `None`. The tests went fine with `None` instead of `{}` and I couldn't find the reason for `{}` over `None` in the couchdb docs.

Reply:
I see this works because 'null' sorts before all other values. But you could also just leave the last element out which is how it was before.

See https://docs.couchdb.org/en/stable/ddocs/views/collation.html#collation-specification.